File: cloud/Server/Controllers/UserController.cs
// Copyright © Spatial Corporation. All rights reserved.

using Spatial.Cloud.Contracts.Users;
using Spatial.Networking;

namespace Spatial.Cloud.Controllers.Users;

/// <summary>
/// A controller for <see cref="User"/>.
/// </summary>
[Module]
[Path("users")]
public class UserController : Controller
{
    /// <summary>
    /// Create a new <see cref="User"/>.
    /// </summary>
    /// <param name="credentials">The user's <see cref="Credentials"/>.</param>
    /// <returns>A <see cref="User"/>.</returns>
    [POST]
    [Path("/")]
    public async Task<User> CreateUserAsync([Body] Credentials credentials)
    {
        // ...

        return await Task.FromResult(new User());
    }
}

File: interface/client/controllers/UserController.ts
// Copyright © Spatial Corporation. All rights reserved.

import { User } from "@sptlco/data";
import { Controller } from "..";

/**
 * A controller for users.
 */
export class UserController extends Controller {
  /**
   * Get the current user.
   * @returns The current user.
   */
  public current = async () => {
    return this.get<User>("users/current");
  };

  /**
   * Get a list of users.
   * @returns A list of users.
   */
  public list = async () => {
    return this.get<User[]>("users/list");
  };
}
